src/axolotl/utils/models.py

```diff
@@ -671,7 +671,7 @@ def load_model(
     if not cfg.fsdp:
         # FSDP doesn't like mixed Float and BFloat16
         for name, module in model.named_modules():
-            if any(m in name for m in ["norm", "gate"]):
+            if "norm" in name or (model_config.model_type == "mixtral" and "gate" in name):
```

Comment:
```suggestion
            if "norm" in name or name == "gate":
```
there are other MoE's out there too. also, `"gate" in name` will still still catch the `gate_proj` module in mixtral.

Reply:
do all MoE's strictly use the same name `gate`? Should we specifically check all those model types?

Reply:
Maybe gate in name and proj not in name?

Reply:
See https://huggingface.co/mlabonne/phixtral-2x2_8/tree/main

Reply:
I think `name == "gate"` would fail as:
```python
for name, module in model.named_modules():
    if "gate" in name:
        print(name)

# model.layers.0.block_sparse_moe.gate
# model.layers.1.block_sparse_moe.gate
..
```

I have also checked, and I could not find `gate_proj` in either the modeling code or after initiating the model.

Reply:
Unresolving back as I think you resolved this while I was midtyping

Reply:
The other suggestion of yours may work: `gate in name and proj not in name`.